.github/workflows/model-evaluation.yml

```diff
@@ -0,0 +1,201 @@
+name: Full Model Evaluation
+
+on:
+  workflow_dispatch:
+    inputs:
+      provision-and-deploy:
+        description: 'Whether to provision the deployment'
+        required: true
+        default: true
+        type: boolean
+      deploy-only:
+        description: 'Whether to deploy the application'
+        required: true
+        default: false
+        type: boolean
```

Comment:
I already changed the logic here, cases like "run provision but no deploy" will not happen. If I change it to an enumeration config with 3 options like this:  `pre-processing: [nothing, deploy-only, provision-deploy]`, would that be better?

Reply:
Did you read my comment?

>  But if the infrastructure exists, the provisioning step will be a noop anyway.

If the Azure Resource Manager detects that what you're trying to provision already exists without changes, it will skip provisioning. There's no need for logic like this here.

Reply:
ok I see, I'll change that it to a single boolean option here.

Reply:
The only option we need is the one controlling resource deletion!